src/main/java/uk/gov/hmcts/dm/service/AzureMediaUploadService.java

```diff
@@ -0,0 +1,295 @@
+package uk.gov.hmcts.dm.service;
+
+import com.microsoft.azure.management.mediaservices.v2018_07_01.*;
+import com.microsoft.azure.management.mediaservices.v2018_07_01.implementation.MediaManager;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.dm.domain.*;
+import uk.gov.hmcts.dm.repository.AmsJobRepository;
+
+import java.net.URI;
+import java.util.*;
+
+@Service
+public class AzureMediaUploadService {
+
+    private static final Logger log = LoggerFactory.getLogger(AzureMediaUploadService.class);
+
+    @Value("${azure.media-services.encodingTransformName}")
+    String encodingTransformName;
+
+    @Value("${azure.media-services.resourcegroup}")
+    private String resourceGroup;
+
+    @Value("${azure.media-services.account.name}")
+    private String accountName;
+
+    @Autowired
+    private MediaManager manager;
+
+    @Autowired
+    private AmsJobRepository amsJobRepository;
+
+    public AmsJob uploadMediaFile(final MultipartFile file,
+                                final DocumentContentVersion documentContentVersion) {
+
+        // Creating a unique suffix so that we don't have name collisions if you run the sample
+        // multiple times without cleaning up.
+        String uniqueness = UUID.randomUUID().toString();
+        String locatorName = "locator-" + uniqueness;
```

Comment:
Would these `xName` variables be better created from methods to better enforce the naming standard? It may help when it comes to writing the retrieval code as well, else this concatenation logic will get duplicated.